### DISCUSSION TITLE
createEffect handler with several arguments

### DISCUSSION THREAD

**Discussion by Aleksey-Danchin** [zerobias/effector#477] [Q&A] [answered]
Hi, guys! For what reason the handler of createEffect has only one argument? I understand the philosophy of `payload arg`, but why use handler with several arguments it's bad idea?

https://share.effector.dev/DXX3HeuG

**Comment by pxbuffer** [comment]
Because it's not extendable and testable. The more args than more painful to test and refactor. Highly recommend to look at [this](https://softwareengineering.stackexchange.com/questions/145055/are-there-guidelines-on-how-many-parameters-a-function-should-accept)

**Comment by zerobias** [comment, marked as answer]
Second argument doesn’t make sense in reactive dataflow. What does it mean to [forward](https://effector.dev/docs/api/effector/forward) to effect with pair of arguments? You need to invent multi-value return at first
